### PR TITLE
fix: env containing http/s not working[#223]

### PR DESCRIPTION
**Pull request by LordNayan**
## Description

This pr fixes the scenario where a env variable contains http/s.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

File: src/lib/components/collections/req-res-section/sub-components/sub-components-header/ApiSendRequestPage.ViewModel.ts [hunk 1]
```diff
@@ -248,9 +248,9 @@ class ApiSendRequestViewModel {
...
     return [
-      this.setEnvironmentVariables(
-        this.extractURl(request.url, request),
-        environmentVariables,
+      this.extractURl(
+        this.setEnvironmentVariables(request.url, environmentVariables),
+        request,
       ),
       this.extractMethod(request.method),
       this.setEnvironmentVariables(
```
Comment (itsmdasifraza) [inline src/lib/components/collections/req-res-section/sub-components/sub-components-header/ApiSendRequestPage.ViewModel.ts:256]: Is env variable working on query params also? 

if I add ?asif={{asif}}

File: src/lib/components/collections/collection-tab/MyCollection.viewModel.ts [hunk 2]
```diff
@@ -29,40 +29,48 @@ export class MyCollectionViewModel {
...
-          { name: newCollectionName },
-        );
+    const updateObj = {};
 
-      tabName = updateCollectionElement.data.data.name;
-
-      collectionsMethods.updateCollection(
+    switch (property) {
+      case "name": {
+        updateObj.name = value;
```
Comment (parthdalaltech) [inline src/lib/components/collections/collection-tab/MyCollection.viewModel.ts:40]: you can make it -> updateObj[property] = value ;

**Review by parthdalaltech — CHANGES_REQUESTED** [review]

File: src/lib/components/collections/req-res-section/sub-components/sub-components-header/ApiSendRequestPage.ViewModel.ts (hunk 1 above)
Comment (LordNayan) [reply src/lib/components/collections/req-res-section/sub-components/sub-components-header/ApiSendRequestPage.ViewModel.ts]: Screenshot 2024-02-14 at 04 30 16

This is done. @itsmdasifraza

**Review by parthdalaltech — APPROVED** [review]

**Review by parthdalaltech — APPROVED** [review]

**Pull request merged**